docs/simple_switch.md

```diff
@@ -311,11 +316,6 @@ if (resubmit was called) {
     for the mcast_grp value.  Enqueue each one in the appropriate
     packet buffer queue.  The instance_type of each will be
     PKT_INSTANCE_TYPE_REPLICATION.
-} else if (egress_spec == DROP_PORT) {
-    // This condition will be true if your code called the
-    // mark_to_drop (P4_16) or drop (P4_14) primitive action during
-    // ingress processing.
-    Drop packet.
```

Comment:
Hi Steffen, I noticed that the ingress thread in [simple_switch.cpp](https://github.com/p4lang/behavioral-model/blob/main/targets/simple_switch/simple_switch.cpp#L619C1-L635C6) implements these conditions as follows:
```C++
    // MULTICAST
    if (mgid != 0) {
      BMLOG_DEBUG_PKT(*packet, "Multicast requested for packet");
      auto &f_instance_type = phv->get_field("standard_metadata.instance_type");
      f_instance_type.set(PKT_INSTANCE_TYPE_REPLICATION);
      multicast(packet.get(), mgid);
      // when doing multicast, we discard the original packet
      continue;
    }

    port_t egress_port = egress_spec;
    BMLOG_DEBUG_PKT(*packet, "Egress port is {}", egress_port);

    if (egress_port == drop_port) {  // drop packet
      BMLOG_DEBUG_PKT(*packet, "Dropping packet at the end of ingress");
      continue;
    }
```
However, the [multicast example](https://github.com/p4lang/tutorials/blob/01ed2c499d88420cff382c5dc9444dbba19067c8/exercises/multicast/solution/multicast.p4#L113) in the tutorials repository demonstrates that `drop()` could be used in the egress pipeline to prune multicast packets.

> I empirically observed that mark_to_drop overrides multicast on BMv2.

Does the observed behaviour occur when mark_to_drop is used in the ingress pipeline?

Reply:
Yes, I was referring to mark_to_drop in the ingress pipeline.